bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,306 @@
+const dbConnection = require('kotoba-node-common').database.connection;
+const WordScheduleModel = require('kotoba-node-common').models.createWordSchedulesModel(dbConnection);
+const showRandomWord = require('./show_random_word.js');
+
+const frequencyCheck = 60 * 1000;
+
+const statusConstants = {
+  running: 'running',
+  paused: 'paused',
+};
+
+/**
+ * Starts the interval
+ * @param {Monochrome} monochrome
+ * @param {boolean} firstCall
+ */
+async function setTimer(monochrome, firstCall) {
```

Comment:
We'll need to handle errors in here, since they won't bubble up to the command framework to get reported. Maybe:

```js
async function setTimer(monochrome, firstCall) {
  let nextTime;
  let now = (new Date()).getTime();
  try {
    // The first time this function is called we don't want to run the command,
    // just set the timer so it runs in sync with the frequencyCheck value
    if (!firstCall) {
      const pendingSchedules = await WordScheduleModel.find({ status: statusConstants.running });
      await Promise.all(pendingSchedules.map(async (schedule) => {
        // We check if the schedule has reached the start date and if it's in sync
        // with the frequency, with an error margin of 0.1s
        if (now >= schedule.start && (now - schedule.start) % schedule.frequency < 100) {
          const channel = getWOTDChannel(
            monochrome.getErisBot(),
            schedule.serverId,
            schedule.id,
          );

          if (channel) {
            await showRandomWord(schedule.level, channel, monochrome);
          } else {
            await WordScheduleModel.deleteOne({ _id: schedule.id });
          }
        }
      }));
    }
    // Since it's an asynchronous function, we sync it manually with the clock for it's next execution
    now = (new Date()).getTime();
    const offset = now % frequencyCheck;
    nextTime = frequencyCheck - offset;
  } catch (err) {
    monochrome.getLogger().error({
      err,
      event: 'WOTD ERROR',
    });
  }

  // If there was an error and nextTime wasn't calculated, try again in 15 minutes
  nextTime = nextTime || (now + (15 * 60 * 1000));

  setTimeout(() => {
    setTimer(monochrome);
  }, nextTime);
}
```